src-electron/db/query-config.js

```diff
@@ -211,15 +212,17 @@ async function insertOrUpdateAttributeState(
       attributeId,
       clusterRef
     )
-  if (
-    staticAttribute.storagePolicy ==
-    dbEnum.storagePolicy.attributeAccessInterface
-  ) {
-    staticAttribute.storagePolicy = dbEnum.storageOption.external
-  } else {
-    staticAttribute.storagePolicy = dbEnum.storageOption.ram
-  }
-
+  let forcedExternal = await queryUpgrade.getForcedExternalStorage(
+    db,
+    attributeId
+  )
+  let storageOption = await queryUpgrade.computeStorageNewConfig(
```

Comment:
In `computeStorageNewConfig` you do `let clusterName
   clusterName = await queryCluster.selectClusterName(db, clusterRef)`

What I mean to say is that you don't need to do that since you have access to `cluster.name` here. Just pass that as an argument and don't do `let clusterName
   clusterName = await queryCluster.selectClusterName(db, clusterRef)`

Reply:
cluster.name is here? the cluster object does not have name

Reply:
```/**
 * Get or inserts default endpoint type cluster given endpoint type, cluster ref, and side.
 * @param {*} db
 * @param {*} endpointTypeId
 * @param {*} clusterRef
 * @param {*} side
 */

async function insertOrSelectDefaultEndpointTypeCluster(
  db,
  endpointTypeId,
  clusterRef,
  side
) {
  await dbApi.dbInsert(
    db,
    `
INSERT OR IGNORE
INTO ENDPOINT_TYPE_CLUSTER (
  ENDPOINT_TYPE_REF, CLUSTER_REF, SIDE, ENABLED
) VALUES ( ?, ?, ?, ? )
`,
    [endpointTypeId, clusterRef, side, false]
  )

  let eptClusterData = await dbApi.dbGet(
    db,
    `
SELECT
  ENDPOINT_TYPE_CLUSTER_ID,
  ENDPOINT_TYPE_REF,
  CLUSTER_REF,
  SIDE,
  ENABLED
FROM ENDPOINT_TYPE_CLUSTER
WHERE ENDPOINT_TYPE_REF = ?
  AND CLUSTER_REF = ?
  AND SIDE = ?`,
    [endpointTypeId, clusterRef, side]
  )

  return dbMapping.map.endpointTypeCluster(eptClusterData)
}